lib/galaxy/schema/jobs.py

```diff
@@ -0,0 +1,268 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+from pydantic import (
+    Extra,
+    Field,
+    Required,
+    UUID4,
+)
+
+from galaxy.schema.fields import (
+    DecodedDatabaseIdField,
+    EncodedDatabaseIdField,
+)
+from galaxy.schema.schema import (
+    DatasetSourceType,
+    EncodedDatasetSourceId,
+    EntityIdField,
+    JobSummary,
+    Model,
+    UuidField,
+)
+
+
+class JobInputSummary(Model):
+    has_empty_inputs: bool = Field(
+        default=Required,
+        title="Empty inputs",
+        description="Job has empty inputs.",
+    )
+    has_duplicate_inputs: bool = Field(
+        default=Required,
+        title="Duplicate inputs",
+        description="Job has duplicate inputs.",
+    )
+
+
+# TODO: Use Tuple again when `make update-client-api-schema` supports them
+class JobErrorSummary(Model):
+    # messages: List[Union[Tuple[str, str], List[str]]]
+    messages: List[List[str]] = Field(
+        default=Required,
+        title="Error messages",
+        description="The error messages for the specified job.",
+    )
+
+
+class JobAssociation(Model):
+    name: str = Field(
+        default=Required,
+        title="name",
+        description="The name of the associated dataset.",
```

Comment:
```suggestion
        description="Name of the job parameter.",
```
This is the input or output name as defined in the tool, the name of the dataset is something different